core/save-profiledata.cpp

```diff
@@ -175,32 +187,214 @@ static std::string format_st_event(const plot_data &entry, const plot_data &next
 	double value;
 	int decimals;
 	const char *unit;
-
-	if (entry.sec < offset || entry.sec > offset + length)
+	if (next_entry.sec < offset || entry.sec > offset + length)
```

Comment:
Why are we relaxing this condition here? Presumably if `entry.sec` is smaller than `offset` we are outside of the video timeline, even if `next_entry.sec` is not smaller than `offset`.

Reply:
It's the fencepost problem. If we have one datapoint at 35:20 and one at 35:40, but the video starts at 35:30, we need to use the data from the 35:20 datapoint for the first 10 seconds of the clip.